Packages/com.unity.inputsystem/InputSystem/API/InputBindingsCollection.cs

```diff
@@ -0,0 +1,203 @@
+            }
+        }
+
+        /// <inheritdoc/>
+        public IEnumerator<BindingWithIndex> GetEnumerator()
+        {
+            return new Enumerator(m_InputAction, m_EnumerationBehaviour);
+        }
+
+        /// <inheritdoc/>
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        private struct Enumerator : IEnumerator<BindingWithIndex>
+        {
+            private readonly EnumerationBehaviour m_EnumerationBehaviour;
+            private readonly ReadOnlyArray<InputBinding> m_Bindings;
+            private readonly InputAction m_InputAction;
+            private int m_Current;
+            private int m_CurrentWithCompositeParts;
+            private int m_InitialBindingsCount;
+
+            public Enumerator(InputAction inputAction, EnumerationBehaviour enumerationBehaviour)
+            {
+                Debug.Assert(inputAction != null);
+
+                m_InputAction = inputAction;
+                m_Bindings = inputAction.bindings;
+                m_EnumerationBehaviour = enumerationBehaviour;
+                m_Current = -1;
+                m_CurrentWithCompositeParts = -1;
+                m_InitialBindingsCount = m_Bindings.Count;
+            }
+
+            public bool MoveNext()
+            {
+                // poor mans enumerator versioning. Note that we have to access
+                // the bindings array from the Input Action here so that it forces it to be recreated if any bindings have
+                // been added or deleted.
+                if (m_InitialBindingsCount != m_InputAction.bindings.Count)
+                    throw new InvalidOperationException("The collection was modified after the enumerator was created.");
+
+                if (m_CurrentWithCompositeParts >= m_Bindings.Count)
+                    return false;
+
+                if (m_EnumerationBehaviour == EnumerationBehaviour.SkipCompositeParts)
+                {
+                    // ++m_Current;
+                    // var i = m_Current;
+                    while (++m_CurrentWithCompositeParts < m_Bindings.Count)
+                    {
+                        if (m_Bindings[m_CurrentWithCompositeParts].isPartOfComposite)
+                            continue;
+
+                        ++m_Current;
+                        return true;
+                    }
+
+                    // if (m_CurrentWithCompositeParts == m_Bindings.Count && m_Bindings[m_CurrentWithCompositeParts - 1].isPartOfComposite)
+                    return false;
+                }
+
+                ++m_CurrentWithCompositeParts;
+                return m_CurrentWithCompositeParts != m_Bindings.Count;
+            }
+
+            public void Reset()
+            {
+                m_Current = -1;
+            }
+
+            public BindingWithIndex Current =>
+                new BindingWithIndex(
```

Comment:
Quite a long line but it does make sense. Would be nice if it could be easier to read